docs/proposals/13-automated-seed-management.md

```diff
@@ -0,0 +1,289 @@
+
+* The `gardenlet` is configured with certain *resources* and their total *capacity* (and, for certain resources, the amount reserved for Gardener).
+* The `gardenlet` seed controller updates the Seed status with the capacity of each resource and how much of it is actually available to be consumed by shoots, using `capacity` and `allocatable` fields that are very similar to the corresponding fields in [the Node status](https://github.com/kubernetes/api/blob/2c3c141c931c0ab1ce1396c3152c72852b3d37ee/core/v1/types.go#L4582-L4593).
+* When scheduling shoots, `gardener-scheduler` is influenced by the remaining capacity of the seed. In the simplest possible implementation, it never schedules shoots onto a seed that has already reached its capacity for a resource requested by the shoot.
+
+Initially, the only resource considered would be the maximum number of shoots that can be scheduled onto a seed. Later, more resources could be added, as well as requests for such resources by shoots. For example, if a shoot requests (via a `resourceRequirements` field) a certain number of persistent volumes, but the seed's allocatable amount of persistent volumes minus the number of persistent volumes already requested by other shoots scheduled onto that seed is lower, the shoot can't be scheduled onto that seed.
+
+In addition, an extensibility plugin framework could be introduced in the future in order to advertise custom resources, including provider-specific resources, that shoots would be able to request, and `gardenlet` would be able to update the seed status with their capacity and allocatable values, for example load balancers on Azure.
+
+Example Seed status with `capacity` and `allocatable` fields:
+
+```yaml
+status:
+  capacity:
+    shoots: "100"
+    persistent-volumes: "200" # Built-in resource
+    azure.provider.extensions.gardener.cloud/load-balancers: "30" # Custom resource advertised by an Azure-specific plugin
+  allocatable:
+    shoots: "100"
+    persistent-volumes: "197" # 3 persistent volumes are reserved for Gardener
+    azure.provider.extensions.gardener.cloud/load-balancers: "300"
+```
+
+Example Shoot `resourceRequirements` field requesting a built-in and a custom resource:
+
+```yaml
+spec:
+  resourceRequirements:
+    requests:
+      persistent-volumes: 20
+      azure.provider.extensions.gardener.cloud/load-balancers: 10
+```
+
+### Gardenlet Configuration
+
+As mentioned above, the total resource capacity for built-in resources such as the number of shoots is specified as part of the `gardenlet` configuration, not in the Seed spec. The `gardenlet` configuration itself could be specified in the spec of the newly introduced [ShootedSeed](#shootedseeds) resource, which also contains the Seed template. Here it is assumed that in the future this could become the recommended and most widely used way to manage seeds. If the same `gardenlet` is responsible for multiple seeds, they would all share the same capacity settings.
```

Comment:
Can we come up with a better name for the resource? When I hear `ShootedSeed` I think about a seed that is under gun fire. Something like  `Root` or `SeedTemplate`?

Reply:
It's also inaccurate, since "shooted seed" would mean "a seed that has shoots", while in our case it's more "seeded shoot" - a shoot that became a seed. However, I would personally rather keep "shooted seed" since we are used to it already, and analogies with actual gardens, seeds, and shoots have their limitations.

Reply:
Renamed to `ManagedSeed` in the latest revision.